test-prow-e2e.sh

```diff
@@ -0,0 +1,136 @@
+  oc get installplan -n "$NAMESPACE" -o yaml >>"${ARTIFACT_DIR}"/installplan.yaml
+  oc get nodes -o yaml >>"${ARTIFACT_DIR}"/node.yaml
+  oc get pods -n "$NAMESPACE" -o yaml >>"${ARTIFACT_DIR}"/pod_details_"$NAMESPACE".yaml
+  for pod in $(oc get pods -n "$NAMESPACE" --no-headers -o custom-columns=":metadata.name" | grep "df-console"); do
+    echo "$pod"
+    oc logs "$pod" -n "$NAMESPACE" >"${ARTIFACT_DIR}"/"${pod}".log
+  done
+  oc get serviceaccounts -n "$NAMESPACE" -o yaml >>"${ARTIFACT_DIR}"/serviceaccount.yaml
+  oc get console.v1.operator.openshift.io cluster -o yaml >>"${ARTIFACT_DIR}"/cluster.yaml
+
+  if [ -d "$ARTIFACT_DIR" ] && [ -d "$SCREENSHOTS_DIR" ]; then
+    if [[ -z "$(ls -A -- "$SCREENSHOTS_DIR")" ]]; then
+      echo "No artifacts were copied."
+    else
+      echo "Copying artifacts from $(pwd)..."
+      cp -r "$SCREENSHOTS_DIR" "${ARTIFACT_DIR}/gui-test-screenshots"
+    fi
+  fi
+}
+
+trap generateLogsAndCopyArtifacts EXIT
+trap generateLogsAndCopyArtifacts ERR
+
+PULL_SECRET_PATH="/var/run/operator-secret/dockerconfig"
+NAMESPACE="openshift-marketplace"
+SECRET_NAME="df-secret"
+ARTIFACT_DIR=${ARTIFACT_DIR:=/tmp/artifacts}
+SCREENSHOTS_DIR=gui-test-screenshots
+
+function createSecret {
+  oc create secret generic "${SECRET_NAME}" --from-file=.dockerconfigjson=${PULL_SECRET_PATH} --type=kubernetes.io/dockerconfigjson -n "$1"
+}
+
+function linkSecrets {
+  for serviceAccount in $(oc get serviceaccounts -n "$NAMESPACE" --no-headers -o custom-columns=":metadata.name" | sed 's/"//g'); do
+    echo "Linking ${SECRET_NAME} to ${serviceAccount}"
+    oc secrets link "${serviceAccount}" "${SECRET_NAME}" -n "$NAMESPACE" --for=pull
+  done
+}
+
+function deleteAllPods {
+  oc delete pods --all -n "$1"
+}
+
+oc patch operatorhub.config.openshift.io/cluster -p='{"spec":{"sources":[{"disabled":true,"name":"redhat-operators"}]}}' --type=merge
+
+echo "Creating secret for CI builds in ${NAMESPACE}"
+createSecret ${NAMESPACE}
+oc apply -f openshift-ci/df-catalog-source.yaml
+
+echo "Waiting for CatalogSource to be Ready"
+# Have to sleep here for atleast 1 min to ensure catalog source is in stable READY state
+sleep 60
+
+echo "Creating secret for linking pods"
+createSecret "$NAMESPACE"
+
+echo "Adding secret to all service accounts in \"$NAMESPACE\" namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods "$NAMESPACE"
+sleep 30
+
+echo "Adding secret to all service accounts in \"$NAMESPACE\" namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods "$NAMESPACE"
+sleep 120
+
+echo "Adding secret to all service accounts in \"$NAMESPACE\" namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods "$NAMESPACE"
+
+echo "Adding secret to all service accounts in \"$NAMESPACE\" namespace"
+linkSecrets
+
+echo "Restarting pods for secret update"
+deleteAllPods "$NAMESPACE"
+sleep 120
```

Comment:
need to test it out once later, I don't think we will need so many `sleep` for Data Federation...

Reply:
Same as ODF, we can test this out later to shave off extra seconds.